### PR TITLE
Replace irange by plate in docs, examples, tests

### LINKED ISSUE #1465: Combine irange + iarange + broadcast into pyro.plate
New users of Pyro have often found it difficult to inderstand the interplay between `iarange`, `irange`, and `@broadcast`. This issue proposes to merge these three concepts into a single `plate`.

Historically @fritzo split `map_data` into `irange`+`iarange`, @neerajprad added `@broadcast`. Now that our usage patterns have converged, @eb8680 suggested making `iarange` and `irange` into messengers, in which case all can be fused.

## Tasks

- [x] #1307 Make irange and iarange into messengers
- [x] #1464 Fuse iarange + broadcast -> plate
- [ ] #1550 Replace irange with plate (requires disambiguating lots of docs & error messages)
- ~~Make plate into a handler? (requires replacing `:class:` with `:func:` in docstrings)~~

### PR DESCRIPTION
**Pull request by fritzo**
Resolves #1465 

This replaces `irange` with `plate` in all docs, examples, and tests:
```diff
- for i in pyro.irange("data", 100):
+ for i in pyro.plate("data", 100):
```
There is no behavior change, since `irange` is already implemented via `plate`.

This PR leaves `irange` in place for backwards compatibility, but adds a deprecation warning when users use `irange`.

### CODE REVIEW

File: tests/infer/test_elbo_mapdata.py [hunk 1]
```diff
@@ -54,8 +54,8 @@ def add_data_point(x, y):
     def model():
         loc_latent = pyro.sample("loc_latent",
                                  dist.Normal(loc0, torch.pow(lam0, -0.5)).independent(1))
-        if map_type == "irange":
-            for i in pyro.irange("aaa", len(data), batch_size):
+        if map_type == "iplate":
+            for i in pyro.iplate("aaa", len(data), batch_size):
```
Comment (neerajprad) [inline tests/infer/test_elbo_mapdata.py]: `iplate` -> `plate`.

File: tests/ops/test_stats.py [hunk 2]
```diff
@@ -9,14 +9,18 @@
 
 def test_autocorrelation():
     x = torch.arange(10.)
-    assert_equal(autocorrelation(x),
+    with xfail_if_not_implemented():
```
Comment (neerajprad) [inline tests/ops/test_stats.py]: Thanks for this fix, also reported in #1548.

Comment (fehiepsi) [reply tests/ops/test_stats.py]: +1

File: tests/poutine/test_mapdata.py [hunk 3]
```diff
@@ -124,22 +124,22 @@ def plate_cuda_model(subsample_size):
         pyro.sample("x", dist.Normal(loc[batch], scale[batch]))
 
 
-def irange_cuda_model(subsample_size):
+def iplate_cuda_model(subsample_size):
     loc = torch.zeros(20).cuda()
     scale = torch.ones(20).cuda()
-    for i in pyro.irange("data", 20, subsample_size, device=loc.device):
+    for i in pyro.iplate("data", 20, subsample_size, device=loc.device):
```
Comment (neerajprad) [inline tests/poutine/test_mapdata.py]: This one is still left, and will break cuda test. :)

**Comment by fritzo** [comment]
Ok, hopefully ready to merge if tests pass :smile:

**Pull request merged**